src/sql-parser/src/parser.rs

```diff
@@ -43,6 +43,14 @@ macro_rules! parser_err {
     };
 }
 
+macro_rules! maybe {
```

Comment:
Is this a standard macro? If not, consider fusing it with `maybe_parse` since they always seem to be called together?

Reply:
Nope, just made it up. It's sort of the inverse of the `try!`/`?` macro—early return if you get the success case rather than the error case.

I really wanted to figure out how to get a `maybe_parse!` macro to work, but I couldn't come up with a design that seemed quite right because of macro hygiene issues. The macro needs to take `self` as input in order to have access to it so it reads kind of weird and is super specific:

```
maybe_parse!(self, |parser| {
    // ...
});
```

But maybe that's preferable! Not sure! Thoughts?

Reply:
Ah I see! I didn't know about the hygiene issues—no strong feelings from me!